models/task.go

```diff
@@ -0,0 +1,47 @@
+package models
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ONSdigital/dp-search-reindex-api/config"
+	"github.com/ONSdigital/dp-search-reindex-api/url"
+)
+
+// Task represents a job metadata model and json representation for API
+type Task struct {
+	JobID             string     `bson:"job_id" json:"job_id"`
+	LastUpdated       time.Time  `bson:"last_updated" json:"last_updated"`
+	Links             *TaskLinks `bson:"links" json:"links"`
+	NumberOfDocuments int        `bson:"number_of_documents" json:"number_of_documents"`
+	TaskName          string     `bson:"task_name" json:"task_name"`
+}
+
+// TaskLinks is a type that contains links to the endpoints for returning a specific task (self), and the job that it is part of (job), respectively.
+type TaskLinks struct {
+	Self string `json:"self"`
+	Job  string `json:"job"`
+}
+
+// NewTask returns a new Task resource that it creates and populates with default values.
+func NewTask(jobID string, taskName string, numDocuments int) (Task, error) {
+	cfg, err := config.Get()
```

Comment:
Consider providing  the bind address as a param. The config is only required to get the bind address I think it reasonable to provide that as a single param. Also if it is a param then you will no longer need to get the config and wont need to return an error from this methods.

Reply:
yes, good idea, I wish I had thought of it